### ISSUE TITLE
[Bug]: segments are not compacted to a reasonable size even after 2+ days

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#24730]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230601-93ea9c49
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2):
```

### Current Behavior

I have 100M-128d rows in the collection, check the segments with birdwathcer: most segments are 240,000 rows, which is only 25% of the segmentMaxSize

### Expected Behavior

most segments compact to 1,000,000 rows

### Steps To Reproduce

```markdown
1. keep search permanently 
2. delete 10% rows at 9am and 9pm everyday, insert 10% rows at 10am and 10pm
3.
```

### Milvus Log

segments info as below, most are around 240,000 rows, and only some are close to 1,000,000 rows
```
Milvus(yanliang-cd-mmszh) > show segment --collection 441397524204357209
SegmentID: 441686852455855394 State: Flushed, Row Count:987108
SegmentID: 441686852455855564 State: Flushed, Row Count:995178
SegmentID: 441686852455855649 State: Flushed, Row Count:1013208
SegmentID: 441686852455855746 State: Flushed, Row Count:1014212
SegmentID: 441686852455856007 State: Flushed, Row Count:1029496
SegmentID: 441686852455856104 State: Flushed, Row Count:1027162
SegmentID: 441686852455856788 State: Flushed, Row Count:1003454
SegmentID: 441686852455856879 State: Flushed, Row Count:1007039
SegmentID: 441686852455857000 State: Flushed, Row Count:1048696
SegmentID: 441686852455857109 State: Flushed, Row Count:1010430
SegmentID: 441686852455857194 State: Flushed, Row Count:1021204
SegmentID: 441686852455857309 State: Flushed, Row Count:1015069
SegmentID: 441686852455857473 State: Flushed, Row Count:994057
SegmentID: 441686852455857576 State: Flushed, Row Count:1020907
SegmentID: 441686852455857625 State: Flushed, Row Count:1074995
SegmentID: 441686852455857716 State: Flushed, Row Count:1075813
SegmentID: 441686852455858585 State: Flushed, Row Count:994204
SegmentID: 441686852455858670 State: Flushed, Row Count:1009504
SegmentID: 441686852455858743 State: Flushed, Row Count:1012342
SegmentID: 441686852455858834 State: Flushed, Row Count:1013960
SegmentID: 441686852455858919 State: Flushed, Row Count:1018826
SegmentID: 441686852455859010 State: Flushed, Row Count:1019607
SegmentID: 441686852455859095 State: Flushed, Row Count:1022762
SegmentID: 441686852455859180 State: Flushed, Row Count:1025757
SegmentID: 441686852455859265 State: Flushed, Row Count:1028736
SegmentID: 441686852455859350 State: Flushed, Row Count:1028957
SegmentID: 441686852455859435 State: Flushed, Row Count:1023819
SegmentID: 441686852455859508 State: Flushed, Row Count:1058193
SegmentID: 441686852455859593 State: Flushed, Row Count:1060576
SegmentID: 441686852455859690 State: Flushed, Row Count:1063995
SegmentID: 441686852455859769 State: Flushed, Row Count:1033359
SegmentID: 441686852455859860 State: Flushed, Row Count:1034145
SegmentID: 441686852455859945 State: Flushed, Row Count:1223212
SegmentID: 441686852455860030 State: Flushed, Row Count:1224033
SegmentID: 441686852455860127 State: Flushed, Row Count:1109192
SegmentID: 441686852455860200 State: Flushed, Row Count:1078715
SegmentID: 441686852455860619 State: Flushed, Row Count:1000000
SegmentID: 441686852455860862 State: Flushed, Row Count:1050000
SegmentID: 441686852455860935 State: Flushed, Row Count:1000000
SegmentID: 441686852455861208 State: Flushed, Row Count:1000000
SegmentID: 441686852455861609 State: Flushed, Row Count:1000000
SegmentID: 441686852455861676 State: Flushed, Row Count:1050000
SegmentID: 441686852455862289 State: Flushed, Row Count:1000000
SegmentID: 441686852455862374 State: Flushed, Row Count:1000000
SegmentID: 441686852455862459 State: Flushed, Row Count:1100000
SegmentID: 441686852455862647 State: Flushed, Row Count:1095872
SegmentID: 441686852455862872 State: Flushed, Row Count:1100000
SegmentID: 441686852455863133 State: Flushed, Row Count:1040000
SegmentID: 441686852455863394 State: Flushed, Row Count:1040000
SegmentID: 441686852455863637 State: Flushed, Row Count:1070000
SegmentID: 441686852455863795 State: Flushed, Row Count:1000000
SegmentID: 441889219770079911 State: Flushed, Row Count:230000
SegmentID: 441889219770280064 State: Flushed, Row Count:230000
SegmentID: 441889219770480197 State: Flushed, Row Count:230000
SegmentID: 441889219770680391 State: Flushed, Row Count:230000
SegmentID: 441889219771080493 State: Flushed, Row Count:230000
SegmentID: 441889219771280602 State: Flushed, Row Count:230000
SegmentID: 441889219771480705 State: Flushed, Row Count:230000
SegmentID: 441889219771680809 State: Flushed, Row Count:230000
SegmentID: 441889219771880913 State: Flushed, Row Count:230000
SegmentID: 441889219772281018 State: Flushed, Row Count:230000
SegmentID: 441889219772481118 State: Flushed, Row Count:230000
SegmentID: 441889219772681220 State: Flushed, Row Count:230000
SegmentID: 441889219772881318 State: Flushed, Row Count:230000
SegmentID: 441889219773281422 State: Flushed, Row Count:230000
SegmentID: 441889219773481526 State: Flushed, Row Count:230000
SegmentID: 441889219773681630 State: Flushed, Row Count:230000
SegmentID: 441889219773881754 State: Flushed, Row Count:240000
SegmentID: 441889219774081950 State: Flushed, Row Count:230000
SegmentID: 441889219774482077 State: Flushed, Row Count:230000
SegmentID: 441889219774682271 State: Flushed, Row Count:230000
SegmentID: 441889219774882450 State: Flushed, Row Count:240000
SegmentID: 441889219775082556 State: Flushed, Row Count:230000
SegmentID: 441889219775482664 State: Flushed, Row Count:220000
SegmentID: 441889219775901219 State: Flushed, Row Count:230000
SegmentID: 441889219776101315 State: Flushed, Row Count:240000
SegmentID: 441889219776301410 State: Flushed, Row Count:230000
SegmentID: 441889219776701503 State: Flushed, Row Count:230000
SegmentID: 441889219776901606 State: Flushed, Row Count:230000
SegmentID: 441889219777101698 State: Flushed, Row Count:230000
SegmentID: 441889219777301792 State: Flushed, Row Count:230000
SegmentID: 441889219777501887 State: Flushed, Row Count:230000
SegmentID: 441889219777901981 State: Flushed, Row Count:230000
SegmentID: 441889219778102068 State: Flushed, Row Count:230000
SegmentID: 441889219778302159 State: Flushed, Row Count:230000
SegmentID: 441889219778502255 State: Flushed, Row Count:230000
SegmentID: 441889219778902345 State: Flushed, Row Count:230000
SegmentID: 441889219779102447 State: Flushed, Row Count:230000
SegmentID: 441889219779302581 State: Flushed, Row Count:230000
SegmentID: 441889219779502729 State: Flushed, Row Count:230000
SegmentID: 441889219779702862 State: Flushed, Row Count:230000
SegmentID: 441889219780103062 State: Flushed, Row Count:230000
SegmentID: 441889219780303158 State: Flushed, Row Count:230000
SegmentID: 441889219780503254 State: Flushed, Row Count:230000
SegmentID: 441889219780703349 State: Flushed, Row Count:230000
SegmentID: 441889219780903444 State: Flushed, Row Count:230000
SegmentID: 441889219781303545 State: Flushed, Row Count:230000
SegmentID: 441889219781503639 State: Flushed, Row Count:230000
SegmentID: 441889219781703733 State: Flushed, Row Count:240000
SegmentID: 441889219781903832 State: Flushed, Row Count:60000
SegmentID: 441889219782322643 State: Flushed, Row Count:240000
SegmentID: 441889219782522735 State: Flushed, Row Count:240000
SegmentID: 441889219782722847 State: Flushed, Row Count:230000
SegmentID: 441889219783123012 State: Flushed, Row Count:230000
SegmentID: 441889219783323103 State: Flushed, Row Count:230000
SegmentID: 441889219783523194 State: Flushed, Row Count:240000
SegmentID: 441889219783723290 State: Flushed, Row Count:240000
SegmentID: 441889219784123403 State: Flushed, Row Count:240000
SegmentID: 441889219784323583 State: Flushed, Row Count:230000
SegmentID: 441889219784523712 State: Flushed, Row Count:230000
SegmentID: 441889219784723894 State: Flushed, Row Count:240000
SegmentID: 441889219785124084 State: Flushed, Row Count:240000
SegmentID: 441889219785324186 State: Flushed, Row Count:230000
SegmentID: 441889219785524284 State: Flushed, Row Count:240000
SegmentID: 441889219785724381 State: Flushed, Row Count:230000
SegmentID: 441889219785924475 State: Flushed, Row Count:230000
SegmentID: 441889219786324572 State: Flushed, Row Count:230000
SegmentID: 441889219786524672 State: Flushed, Row Count:240000
SegmentID: 441889219786724775 State: Flushed, Row Count:240000
SegmentID: 441889219786924874 State: Flushed, Row Count:230000
SegmentID: 441889219787324972 State: Flushed, Row Count:230000
SegmentID: 441889219787525070 State: Flushed, Row Count:230000
SegmentID: 441889219787725180 State: Flushed, Row Count:240000
SegmentID: 441889219787925433 State: Flushed, Row Count:230000
SegmentID: 441889219788325616 State: Flushed, Row Count:240000
SegmentID: 441889219788525788 State: Flushed, Row Count:240000
SegmentID: 441889219788726034 State: Flushed, Row Count:230000
SegmentID: 441889219788926172 State: Flushed, Row Count:70000
SegmentID: 441889219789344032 State: Flushed, Row Count:230000
SegmentID: 441889219789544122 State: Flushed, Row Count:230000
SegmentID: 441889219789744212 State: Flushed, Row Count:230000
SegmentID: 441889219790144303 State: Flushed, Row Count:240000
SegmentID: 441889219790344401 State: Flushed, Row Count:230000
SegmentID: 441889219790544496 State: Flushed, Row Count:230000
SegmentID: 441889219790744592 State: Flushed, Row Count:240000
SegmentID: 441889219790944698 State: Flushed, Row Count:230000
SegmentID: 441889219791344789 State: Flushed, Row Count:230000
SegmentID: 441889219791544878 State: Flushed, Row Count:240000
SegmentID: 441889219791744989 State: Flushed, Row Count:240000
SegmentID: 441889219791945085 State: Flushed, Row Count:230000
SegmentID: 441889219792345174 State: Flushed, Row Count:240000
SegmentID: 441889219792545278 State: Flushed, Row Count:230000
SegmentID: 441889219792745372 State: Flushed, Row Count:230000
SegmentID: 441889219792945479 State: Flushed, Row Count:230000
SegmentID: 441889219793345624 State: Flushed, Row Count:240000
SegmentID: 441889219793545768 State: Flushed, Row Count:230000
SegmentID: 441889219793745928 State: Flushed, Row Count:230000
SegmentID: 441889219793946086 State: Flushed, Row Count:240000
SegmentID: 441889219794146178 State: Flushed, Row Count:240000
SegmentID: 441889219794546272 State: Flushed, Row Count:230000
SegmentID: 441889219794746369 State: Flushed, Row Count:240000
SegmentID: 441889219794946470 State: Flushed, Row Count:240000
SegmentID: 441889219795146574 State: Flushed, Row Count:230000
SegmentID: 441889219795546675 State: Flushed, Row Count:230000
SegmentID: 441889219795746769 State: Flushed, Row Count:240000
SegmentID: 441889219795946877 State: Flushed, Row Count:230000
SegmentID: 441889219796146975 State: Flushed, Row Count:230000
SegmentID: 441889219796547077 State: Flushed, Row Count:230000
SegmentID: 441889219796747177 State: Flushed, Row Count:50000
SegmentID: 441889219796965673 State: Flushed, Row Count:230000
SegmentID: 441889219797365767 State: Flushed, Row Count:240000
SegmentID: 441889219797565861 State: Flushed, Row Count:230000
SegmentID: 441889219797765948 State: Flushed, Row Count:240000
SegmentID: 441889219797966052 State: Flushed, Row Count:240000
SegmentID: 441889219798366149 State: Flushed, Row Count:240000
SegmentID: 441889219798566239 State: Flushed, Row Count:240000
SegmentID: 441889219798766338 State: Flushed, Row Count:230000
SegmentID: 441889219798966429 State: Flushed, Row Count:230000
SegmentID: 441889219799166527 State: Flushed, Row Count:230000
SegmentID: 441889219799566647 State: Flushed, Row Count:230000
SegmentID: 441889219799766811 State: Flushed, Row Count:240000
SegmentID: 441889219799966916 State: Flushed, Row Count:240000
SegmentID: 441889219800167088 State: Flushed, Row Count:240000
SegmentID: 441889219800567448 State: Flushed, Row Count:230000
SegmentID: 441889219800767559 State: Flushed, Row Count:230000
SegmentID: 441889219800967651 State: Flushed, Row Count:240000
SegmentID: 441889219801167742 State: Flushed, Row Count:230000
SegmentID: 441889219801567843 State: Flushed, Row Count:230000
SegmentID: 441889219801767937 State: Flushed, Row Count:230000
SegmentID: 441889219801968029 State: Flushed, Row Count:230000
SegmentID: 441889219802168122 State: Flushed, Row Count:240000
SegmentID: 441889219802368220 State: Flushed, Row Count:230000
SegmentID: 441889219802768311 State: Flushed, Row Count:230000
SegmentID: 441889219802968408 State: Flushed, Row Count:240000
SegmentID: 441889219803168507 State: Flushed, Row Count:230000
SegmentID: 441889219803368605 State: Flushed, Row Count:230000
SegmentID: 441889219803768699 State: Flushed, Row Count:240000
SegmentID: 441889219803968803 State: Flushed, Row Count:230000
SegmentID: 441889219804168906 State: Flushed, Row Count:230000
SegmentID: 441889219804369008 State: Flushed, Row Count:230000
SegmentID: 441889219804569114 State: Flushed, Row Count:230000
SegmentID: 441889219804969211 State: Flushed, Row Count:230000
SegmentID: 441889219805169310 State: Flushed, Row Count:50000
SegmentID: 441889219805386980 State: Flushed, Row Count:230000
SegmentID: 441889219805787188 State: Flushed, Row Count:240000
SegmentID: 441889219805987281 State: Flushed, Row Count:240000
SegmentID: 441889219806187377 State: Flushed, Row Count:230000
SegmentID: 441889219806387467 State: Flushed, Row Count:240000
SegmentID: 441889219806787563 State: Flushed, Row Count:230000
SegmentID: 441889219806987653 State: Flushed, Row Count:240000
SegmentID: 441889219807187751 State: Flushed, Row Count:230000
SegmentID: 441889219807387840 State: Flushed, Row Count:230000
SegmentID: 441889219807787929 State: Flushed, Row Count:230000
SegmentID: 441889219807988016 State: Flushed, Row Count:230000
SegmentID: 441889219808188113 State: Flushed, Row Count:230000
SegmentID: 441889219808388202 State: Flushed, Row Count:240000
SegmentID: 441889219808588301 State: Flushed, Row Count:230000
SegmentID: 441889219808988391 State: Flushed, Row Count:230000
SegmentID: 441889219809188518 State: Flushed, Row Count:240000
SegmentID: 441889219809388783 State: Flushed, Row Count:240000
SegmentID: 441889219809588997 State: Flushed, Row Count:230000
SegmentID: 441889219809989145 State: Flushed, Row Count:240000
SegmentID: 441889219810189237 State: Flushed, Row Count:240000
SegmentID: 441889219810389330 State: Flushed, Row Count:230000
SegmentID: 441889219810589422 State: Flushed, Row Count:240000
SegmentID: 441889219810989517 State: Flushed, Row Count:240000
SegmentID: 441889219811189613 State: Flushed, Row Count:230000
SegmentID: 441889219811389702 State: Flushed, Row Count:230000
SegmentID: 441889219811589794 State: Flushed, Row Count:240000
SegmentID: 441889219811989892 State: Flushed, Row Count:230000
SegmentID: 441889219812189982 State: Flushed, Row Count:230000
SegmentID: 441889219812390074 State: Flushed, Row Count:230000
SegmentID: 441889219812590188 State: Flushed, Row Count:230000
SegmentID: 441889219812790344 State: Flushed, Row Count:230000
SegmentID: 441889219813190518 State: Flushed, Row Count:230000
SegmentID: 441889219813390655 State: Flushed, Row Count:240000
SegmentID: 441889219813590882 State: Flushed, Row Count:240000
SegmentID: 441889219813791008 State: Flushed, Row Count:240000
SegmentID: 441889219814191105 State: Flushed, Row Count:240000
SegmentID: 441889219814391208 State: Flushed, Row Count:100000
SegmentID: 441889219814809272 State: Flushed, Row Count:230000
SegmentID: 441889219815009369 State: Flushed, Row Count:230000
SegmentID: 441889219815209460 State: Flushed, Row Count:230000
SegmentID: 441889219815409551 State: Flushed, Row Count:230000
SegmentID: 441889219815609658 State: Flushed, Row Count:230000
SegmentID: 441889219816009821 State: Flushed, Row Count:240000
SegmentID: 441889219816210026 State: Flushed, Row Count:230000
SegmentID: 441889219816410125 State: Flushed, Row Count:240000
SegmentID: 441889219816610225 State: Flushed, Row Count:230000
SegmentID: 441889219817010334 State: Flushed, Row Count:230000
SegmentID: 441889219817210477 State: Flushed, Row Count:230000
SegmentID: 441889219817410641 State: Flushed, Row Count:230000
SegmentID: 441889219817610921 State: Flushed, Row Count:230000
SegmentID: 441889219817811080 State: Flushed, Row Count:230000
SegmentID: 441889219818211174 State: Flushed, Row Count:230000
SegmentID: 441889219818411271 State: Flushed, Row Count:230000
SegmentID: 441889219818611369 State: Flushed, Row Count:240000
SegmentID: 441889219818811468 State: Flushed, Row Count:230000
SegmentID: 441889219819211561 State: Flushed, Row Count:230000
SegmentID: 441889219819411655 State: Flushed, Row Count:240000
SegmentID: 441889219819611755 State: Flushed, Row Count:230000
SegmentID: 441889219819812034 State: Flushed, Row Count:230000
SegmentID: 441889219820012204 State: Flushed, Row Count:230000
SegmentID: 441889219820412315 State: Flushed, Row Count:240000
SegmentID: 441889219820612444 State: Flushed, Row Count:240000
SegmentID: 441889219820812924 State: Flushed, Row Count:230000
SegmentID: 441889219821013061 State: Flushed, Row Count:240000
SegmentID: 441889219821413157 State: Flushed, Row Count:250000
SegmentID: 441889219821613259 State: Flushed, Row Count:230000
SegmentID: 441889219821813352 State: Flushed, Row Count:230000
SegmentID: 441889219822013451 State: Flushed, Row Count:230000
SegmentID: 441889219822413547 State: Flushed, Row Count:230000
SegmentID: 441889219822613641 State: Flushed, Row Count:230000
SegmentID: 441889219822813738 State: Flushed, Row Count:230000
SegmentID: 441889219823013838 State: Flushed, Row Count:230000
SegmentID: 441889219823213940 State: Flushed, Row Count:240000
SegmentID: 441889219823614048 State: Flushed, Row Count:230000
SegmentID: 441889219823814167 State: Flushed, Row Count:240000
SegmentID: 441889219824014326 State: Flushed, Row Count:230000
SegmentID: 441889219824214566 State: Flushed, Row Count:240000
SegmentID: 441889219824614706 State: Flushed, Row Count:70000
SegmentID: 441889219824831082 State: Flushed, Row Count:230000
SegmentID: 441889219825031173 State: Flushed, Row Count:240000
SegmentID: 441889219825431266 State: Flushed, Row Count:230000
SegmentID: 441889219825631359 State: Flushed, Row Count:240000
SegmentID: 441889219825831457 State: Flushed, Row Count:230000
SegmentID: 441889219826031560 State: Flushed, Row Count:240000
SegmentID: 441889219826431702 State: Flushed, Row Count:240000
SegmentID: 441889219826631911 State: Flushed, Row Count:230000
SegmentID: 441889219826832080 State: Flushed, Row Count:230000
SegmentID: 441889219827032257 State: Flushed, Row Count:240000
SegmentID: 441889219827432363 State: Flushed, Row Count:240000
SegmentID: 441889219827632469 State: Flushed, Row Count:240000
SegmentID: 441889219827832578 State: Flushed, Row Count:60000
SegmentID: 441889219828252745 State: Flushed, Row Count:240000
SegmentID: 441889219828452855 State: Flushed, Row Count:230000
SegmentID: 441889219828653150 State: Flushed, Row Count:230000
SegmentID: 441889219828853251 State: Flushed, Row Count:90000
--- Growing: 0, Sealed: 0, Flushed: 288
--- Total Segments: 288, row count: 106937794
```

**Comment by yanliang567** [comment]
pod names:
```
yanliang-cd-mmszh-milvus-datanode-5575c54b8c-5mw9b                1/1     Running       0               5d22h   10.102.5.16     devops-node21   <none>           <none>
yanliang-cd-mmszh-milvus-datanode-5575c54b8c-zv829                1/1     Running       0               5d22h   10.102.9.157    devops-node13   <none>           <none>
yanliang-cd-mmszh-milvus-indexnode-86989c4586-pg4bb               1/1     Running       0               5d22h   10.102.9.159    devops-node13   <none>           <none>
yanliang-cd-mmszh-milvus-indexnode-86989c4586-t2nf4               1/1     Running       0               5d22h   10.102.6.136    devops-node10   <none>           <none>
yanliang-cd-mmszh-milvus-mixcoord-55f6f69dbc-w5slq                1/1     Running       0               5d22h   10.102.9.171    devops-node13   <none>           <none>
yanliang-cd-mmszh-milvus-proxy-65b64bdf8f-zm5gp                   1/1     Running       0               5d22h   10.102.9.166    devops-node13   <none>           <none>
yanliang-cd-mmszh-milvus-querynode-56cf4f8779-66dnl               1/1     Running       0               5d22h   10.102.7.188    devops-node11   <none>           <none>
yanliang-cd-mmszh-milvus-querynode-56cf4f8779-d2b86               1/1     Running       1 (2d11h ago)   5d22h   10.102.10.131   devops-node20   <none>           <none>
yanliang-cd-mmszh-milvus-querynode-56cf4f8779-fxfk5               1/1     Running       0               5d22h   10.102.6.134    devops-node10   <none>           <none>
yanliang-cd-mmszh-milvus-querynode-56cf4f8779-gpxzl               1/1     Running       0               5d22h   10.102.9.172    devops-node13   <none>           <none>
yanliang-cd-mmszh-milvus-querynode-56cf4f8779-rgfcc               1/1     Running       1 (2d17h ago)   5d22h   10.102.5.15     devops-node21   <none>           <none>
yanliang-cd-mmszh-milvus-querynode-56cf4f8779-s4wrz               1/1     Running       0               5d22h   10.102.9.158    devops-node13   <none>           <none>

```

**Comment by XuanYang-cn** [comment]
same as #24834 
should be fixed by #24835 
/unassign
/assign @yanliang567

**Comment by yanliang567** [comment]
seems working on master-20230619-a6310050. close for now
